source/extensions/filters/network/zookeeper_proxy/zookeeper_filter.cc

```diff
@@ -0,0 +1,239 @@
+  config_->scope_.counter(fmt::format("{}.auth.{}_rq", config_->stat_prefix_, scheme)).inc();
+  setDynamicMetadata("opname", "auth");
+}
+
+void ZooKeeperFilter::onGetDataRequest(const std::string& path, const bool watch) {
+  config_->stats_.getdata_rq_.inc();
+  setDynamicMetadata({{"opname", "getdata"}, {"path", path}, {"watch", watch ? "true" : "false"}});
+}
+
+void ZooKeeperFilter::onCreateRequest(const std::string& path, const CreateFlags flags,
+                                      const OpCodes opcode) {
+  std::string opname;
+
+  switch (opcode) {
+  case OpCodes::CREATE:
+    opname = "create";
+    config_->stats_.create_rq_.inc();
+    break;
+  case OpCodes::CREATE2:
+    opname = "create2";
+    config_->stats_.create2_rq_.inc();
+    break;
+  case OpCodes::CREATECONTAINER:
+    opname = "createcontainer";
+    config_->stats_.createcontainer_rq_.inc();
+    break;
+  case OpCodes::CREATETTL:
+    opname = "createttl";
+    config_->stats_.createttl_rq_.inc();
+    break;
+  default:
+    throw EnvoyException(fmt::format("Unknown opcode: {}", enumToIntSigned(opcode)));
+    break;
+  }
+
+  setDynamicMetadata(
+      {{"opname", opname}, {"path", path}, {"create_type", createFlagsToString(flags)}});
+}
+
+void ZooKeeperFilter::onSetRequest(const std::string& path) {
+  config_->stats_.setdata_rq_.inc();
+  setDynamicMetadata({{"opname", "setdata"}, {"path", path}});
+}
+
+void ZooKeeperFilter::onGetChildrenRequest(const std::string& path, const bool watch,
+                                           const bool v2) {
+  std::string opname = "getchildren";
+
+  if (v2) {
+    config_->stats_.getchildren2_rq_.inc();
+    opname = "getchildren2";
+  } else {
+    config_->stats_.getchildren_rq_.inc();
+  }
+
+  setDynamicMetadata({{"opname", opname}, {"path", path}, {"watch", watch ? "true" : "false"}});
+}
+
+void ZooKeeperFilter::onDeleteRequest(const std::string& path, const int32_t version) {
+  config_->stats_.remove_rq_.inc();
+  setDynamicMetadata({{"opname", "remove"}, {"path", path}, {"version", std::to_string(version)}});
+}
+
+void ZooKeeperFilter::onExistsRequest(const std::string& path, const bool watch) {
+  config_->stats_.exists_rq_.inc();
+  setDynamicMetadata({{"opname", "exists"}, {"path", path}, {"watch", watch ? "true" : "false"}});
+}
+
+void ZooKeeperFilter::onGetAclRequest(const std::string& path) {
+  config_->stats_.getacl_rq_.inc();
+  setDynamicMetadata({{"opname", "getacl"}, {"path", path}});
+}
+
+void ZooKeeperFilter::onSetAclRequest(const std::string& path, const int32_t version) {
+  config_->stats_.setacl_rq_.inc();
+  setDynamicMetadata({{"opname", "setacl"}, {"path", path}, {"version", std::to_string(version)}});
+}
+
+void ZooKeeperFilter::onSyncRequest(const std::string& path) {
+  config_->stats_.sync_rq_.inc();
+  setDynamicMetadata({{"opname", "sync"}, {"path", path}});
+}
+
+void ZooKeeperFilter::onCheckRequest(const std::string&, const int32_t) {
+  config_->stats_.check_rq_.inc();
```

Comment:
Do you not want to set metadata for this one?

Reply:
Check only happens within a multi transaction, and we only emit metadata from `onMulti()` for now. I'll revisit this once I get access log support, in a follow-up PR ;-)